deploy/docker/startup.sh

```diff
@@ -26,6 +29,18 @@ fi
 rmdir /srv/backup
 ln -s ${MNT_DIR}/backup /srv/backup
 
+# Replace the /var/log/apache2 in the image with a remotely-mounted one
+# Add a per-container piece because logs may be written during deployments
+APACHE_LOG_DIR="${MNT_DIR}/apache_logs/${HOSTNAME_LOCALIP_PART}"
+if [ ! -e "${APACHE_LOG_DIR}" ]; then
+  mkdir -p ${APACHE_LOG_DIR}
+  chown root:adm ${APACHE_LOG_DIR}
+  chmod 750 ${APACHE_LOG_DIR}
+fi
+mv -f /var/log/apache2/* ${APACHE_LOG_DIR}/
+rmdir /var/log/apache2
+ln -s ${APACHE_LOG_DIR} /var/log/apache2
```

Comment:
So, in this process of moving, removing the directory, and creating the symbolic link, is there a chance that Apache may wish to log something happening? And if it doesn't find a directory to log to, what happens then? Does it create a directory, which would cause the symbolic linking to fail? Or does it fail silently? Or something else?

Reply:
It looks like ln -Ff might be able to do what we need to handle the case where the directory already exists.

Reply:
That shouldn't be a problem --- this script is run at container startup, and apache isn't running yet.  If you look at the rest of deploy/docker/startup.sh, you'll see that apache gets started later, around line 60.  The only reason logfiles exist is that they're created when apache is installed as part of docker image creation; they shouldn't be being written to at this point.

Reply:
Very good, approving this now.